react-native-keys.podspec

```diff
@@ -14,7 +15,28 @@ Pod::Spec.new do |s|
   s.source       = { :git => "https://github.com/numandev1/react-native-keys.git", :tag => "#{s.version}" }
 
   s.source_files = "ios/**/*.{h,m,mm}", "cpp/**/*.{h,c,cpp}"
+  # Use install_modules_dependencies helper to install the dependencies if React Native version >=0.71.0.
+  # See https://github.com/facebook/react-native/blob/febf6b7f33fdb4904669f99d795eba4c0f95d7bf/scripts/cocoapods/new_architecture.rb#L79.
+  if respond_to?(:install_modules_dependencies, true)
+    install_modules_dependencies(s)
+  else
+  s.dependency "React-Core"
 
+  # Don't install the dependencies when we run `pod install` in the old architecture.
+  if ENV['RCT_NEW_ARCH_ENABLED'] == '1' then
+    s.compiler_flags = folly_compiler_flags + " -DRCT_NEW_ARCH_ENABLED=1"
+    s.pod_target_xcconfig    = {
+        "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/boost\"",
+        "OTHER_CPLUSPLUSFLAGS" => "-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1",
+        "CLANG_CXX_LANGUAGE_STANDARD" => "c++17"
+    }
+    s.dependency "React-Codegen"
```

Comment:
I confirm that this file block is not executing
you can add this `puts "executing RCT_NEW_ARCH_ENABLED"` line in this block to print if this block is executing or not

this command should work, currently, only this command is working but it is also not executing that block `RCT_NEW_ARCH_ENABLED=1 bundle exec pod install`
```
RCT_NEW_ARCH_ENABLED=1 && pod install
```

Reply:
u want to remove this: ``` s.dependency "React-Codegen" ``` ?

Reply:
@MasonLe2497 No, I see `install_modules_dependencies` does not exist in `0.70` we have to just verify if `RCT_NEW_ARCH_ENABLED=1 && pod install` command is enabling new arch or not

Reply:
but `RCT_NEW_ARCH_ENABLED=1 && pod install` this command is not installing new arch, can you check?

Reply:
u can follow [](https://github.com/callstack/react-native-builder-bob/blob/main/packages/create-react-native-library/templates/native-common/%7B%25-%20project.identifier%20%25%7D.podspec)

Reply:
@MasonLe2497 Yes our code matches with the template but we have to fix this code as `RCT_NEW_ARCH_ENABLED=1 && pod install` not installing new arch, we have to verify this command should install new arch

Reply:
you mean: RCT_NEW_ARCH_ENABLED=1 && pod install => install_modules_dependencies not running?

Reply:
@MasonLe2497 no i mean, `cd ios && RCT_NEW_ARCH_ENABLED=1 && pod install` should enable new arch, currently this command is not enabling new arch, only `bundle exec` command is enabling new arch

Reply:
No. the correct command is: 
```sh
cd ios && RCT_NEW_ARCH_ENABLED=1 pod install 
```
if u have ```  &&  ``` syntax, ``` RCT_NEW_ARCH_ENABLED=1 ``` will be a sh command, not change env for cocoapods

Reply:
yes @MasonLe2497 you are right, correct command is `cd ios && RCT_NEW_ARCH_ENABLED=1 pod install`, this command should enable new arch

Reply:
we can resolve this conversation?

Reply:
@MasonLe2497 did you fix enabling new arch by this command `cd ios && RCT_NEW_ARCH_ENABLED=1 pod install`?

Reply:
it still working as well

Reply:
oh. on my side. it working 
<img width="948" alt="image" src="https://github.com/numandev1/react-native-keys/assets/43195241/788c44d5-88fb-4acf-9bb2-7c8e05987fe3">
## Result of rn info:
System:
  OS: macOS 13.4.1
  CPU: (8) arm64 Apple M1
  Memory: 96.25 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 16.20.1
    path: ~/Library/Caches/fnm_multishells/6573_1691481597947/bin/node
  Yarn:
    version: 1.22.19
    path: /opt/homebrew/bin/yarn
  npm:
    version: 8.19.4
    path: ~/Library/Caches/fnm_multishells/6573_1691481597947/bin/npm
  Watchman:
    version: 2023.07.10.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.12.1
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
  Android SDK:
    Android NDK: 22.1.7171670
IDEs:
  Android Studio: 2022.2 AI-222.4459.24.2221.10121639
  Xcode:
    version: 14.3.1/14E300c
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.20
    path: /usr/bin/javac
  Ruby:
    version: 3.2.2
    path: /Users/ngoc.le/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: true